include/tc/lang/sema.h

```diff
@@ -601,5 +613,7 @@ struct Sema {
   // if you write to an input, using it in a range expression is no longer
   // allowed
   std::unordered_set<std::string> live_input_names;
+
+  std::unordered_set<std::string> non_temporaries;
```

Comment:
can you make it cameCase i.e. `nonTemporaries` ? this is as per our coding conventions so we should follow them as well :)